lib/rubocop/cop/style/case_equality.rb

```diff
@@ -47,6 +54,18 @@ def const?(node)
             true
           end
         end
+
+        def replacement(lhs, rhs)
+          case lhs.type
+          when :regexp
+            "#{rhs.source} =~ #{lhs.source}"
+          when :begin
+            child = lhs.children.first
+            "#{lhs.source}.include?(#{rhs.source})" if child&.range_type?
+          when :const
+            "#{rhs.source}.is_a?(#{lhs.source})"
```

Comment:
Not all constants are class/module instances.  Regexps are often bound to a constant, to be used with `===`.

I think a better bet is to replace the constant only if it is CamelCased.

Reply:
Thank you for your feedback. I've opened the PR #8526.